xbmc/utils/StringUtils.cpp

```diff
@@ -1237,6 +1286,51 @@ int StringUtils::AlphaNumericCollation(int nKey1, const void* pKey1, int nKey2,
   if (r == 0)
     return nKey1 - nKey2;
 
+#ifdef TARGET_ANDROID
+  if (g_langInfo.UseLocaleCollation())
+  {
+    UErrorCode ustatus = U_ZERO_ERROR;
+    try
+    {
+      auto ucoll = std::unique_ptr<UCollator, decltype(&ucol_close)>(
+          ucol_open(g_langInfo.GetISOLocale().c_str(), &ustatus), &ucol_close);
+      if (U_FAILURE(ustatus))
+        throw std::runtime_error("StringUtils: ucol_open failed.");
+      ucol_setAttribute(ucoll.get(), UCOL_NUMERIC_COLLATION, UCOL_ON, &ustatus);
+      if (U_FAILURE(ustatus))
+        throw std::runtime_error("StringUtils: failed to set UCOL_NUMERIC_COLLATION.");
+
+      std::vector<UChar> uleft(nKey1 + 1);
```

Comment:
this will most certainly not work, since converting from utf-8 to utf-16 will double the size at least. did you run this code to check?
please also use preflighting here.

Reply:
@yol I think each UTF-16 `char16_t` or `UChar` should come from 1 or more UTF-8 bytes. If nKey is number of bytes, the size should work because `UChar` is of 2 bytes. I can't remember how I tested this though. Anyway, I'm rewriting using preflighting .

Reply:
I think you're right, I might have confused bytes and characters here. But no matter anymore :)